src/campaigns/campaigns.controller.ts

```diff
@@ -78,4 +88,24 @@ export class CampaignsController {
 
     return this.campaignsService.updateJsonFields(campaign.id, body)
   }
+
+  @Post('launch') // campaign/launch.js
+  @UseCampaign()
+  @HttpCode(HttpStatus.OK)
+  async launch(@ReqUser() user: User, @ReqCampaign() campaign: Campaign) {
+    try {
+      return await this.campaignsService.launch(user, campaign)
+    } catch (e) {
+      this.logger.error('Error at campaign launch', e)
+      await this.slack.errorMessage('Error at campaign launch', e)
+
+      if (e instanceof Error) {
+        throw new BadRequestException(
+          'Cannot launch campaign without an Office',
+        )
+      }
+
+      throw e
+    }
+  }
```

Comment:
I don't think this is what we want here.. All exceptions from the service are going to be instances of prototypes of `Error`:
<img width="564" alt="image" src="https://github.com/user-attachments/assets/0e05c5c3-b69b-40b9-8501-9b17ab8b6cfa" />

Prolly better to just throw the `BadRequestException` down in the service, here:  https://github.com/thegoodparty/gp-api/pull/41/files#diff-ffce57cc5ae4181953538cc126a97baa019723136df9a25e9d94154b56b639e8R215  ...and delete this if statement all together.

Reply:
Oh.. you'll def still have to `throw` the exception again so that Nest can handle it accordingly tho..